client/src/components/HistoryViewer/HistoryViewerVersionList.js

```diff
@@ -1,9 +1,27 @@
 import React, { PropTypes, PureComponent } from 'react';
 import classnames from 'classnames';
+import i18n from 'i18n';
+import { clearMessages } from 'state/historyviewer/HistoryViewerActions';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 import { inject } from 'lib/Injector';
+import { messageType } from 'types/messageType';
 import { versionType } from 'types/versionType';
 
 class HistoryViewerVersionList extends PureComponent {
+  /**
+   * Clear any messages so they aren't re-rendered again later. This is controlled by a
+   * boolean property, since the list is also used in the detail context where we don't
+   * want to clear messages from.
+   */
+  componentWillUnmount() {
+    const { onClearMessages, shouldClearMessages } = this.props;
+
+    if (shouldClearMessages) {
+      onClearMessages();
+    }
+  }
```

Comment:
This `shouldClearMessages` prop makes me really uneasy. You're basically hijacking a prop, which is intended to be compositional, and making it procedural. Almost like `myComponent.clearMessages()`.

I feel like there's probably a better approach. Can you help me understand what the intended effect is here? Maybe the clearing of messages should be a side-effect of router state?

Reply:
Sure - so the component should render any messages from the redux state, once. After rendering them it should clear them again. I had some trouble getting the right logic for this without causing infinite re-rendering loops in React =/